pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -84,6 +84,24 @@ const (
 	TemplateNAALabel      = "volume.csi.k8s.io/affinity-source-naa"
 )
 
+var labelValueSanitizer = regexp.MustCompile(`[^A-Za-z0-9\-_.]`)
```

Comment:
didn't we have colons `:`  in that naa as well?
I don't understand how all those chars are part of naa to begin with. 
Couuld it be that we have a bug in the query?

Reply:
According to the bug report the bad naa was something like `mpx.vmhba0:00:00:00`  (I replaced the real chars with zeros)